src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/DictionaryLayoutNode.cs

```diff
@@ -85,6 +203,8 @@ class EntryHashTable : LockFreeReaderHashtable<GenericLookupResult, GenericLooku
         private EntryHashTable _entries = new EntryHashTable();
         private volatile GenericLookupResult[] _layout;
 
+        public override bool HasFixedSlots => false;
+
```

Comment:
I'd like you to look into making ComputeLayout on this type fail if its called to early instead of relying on an assert that EnsureEntry isn't called after ComputeLayout. There is a non-explained implementation detail hanging out here, and I'd prefer it to be solved sooner rather than later.